pkg/common/workbuilder/workbuilder.go

```diff
@@ -0,0 +1,355 @@
+package workbuilder
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"open-cluster-management.io/addon-framework/pkg/common/workapplier"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+// DefaultManifestLimit is the max size of manifests data which is 50k bytes by default.
+const DefaultManifestLimit = 50 * 1024
+
+// DefaultManifestThreshold is the threshold size of manifests for the first new created manifestWork.
+const DefaultManifestThreshold = 0.8
+
+type manifestKey struct {
+	gvk             schema.GroupVersionKind
+	namespace, name string
+}
+
+type manifestSize struct {
+	manifest workapiv1.Manifest
+	size     int
+}
+
+type manifestWorkBuffer struct {
+	work   *workapiv1.ManifestWork
+	buffer int
```

Comment:
could you add some comments for the field `buffer`?

Reply:
added